### PR TITLE
Fhac 141 - Connect/Readme.md updates

### PR DESCRIPTION
**Pull request by sailajaa**
1. Corrected broken history.md link
2. Provided references for all connect supported application servers setup.
3. Content modification under build from source steps.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,10 +27,14 @@ Three primary elements make up the CONNECT solution:
 
 History
 -------
+* 4.4 released December 2014
+* 4.3 released March 2014
+* 4.2 released
+* 4.1 released
```
Comment (christophermay07) [inline README.md]: Do we not have dates for these two releases?  (or does it make more sense to remove this information and use the wiki + HISTORY.md for this?)

File: README.md [hunk 2]
```diff
@@ -27,10 +27,14 @@ Three primary elements make up the CONNECT solution:
...
 * 4.0 released February 2013
 * 3.3 released March 2012
 
-For more information, about CONNECT's history see [HISTORY.md](./CONNECT/HISTORY.md)
+For more information, about CONNECT's history see [HISTORY.md](./HISTORY.md)
```
Comment (christophermay07) [inline README.md]: The comma should be after "history", not "information"

**Comment by christophermay07** [comment]
It doesn't show in the diff on GitHub, but the HIEOS profile should be removed for README.md.

File: README.md [hunk 3]
```diff
@@ -64,7 +71,7 @@ Everyone:
         $ mvn clean install
 
 ####Building an ear
-All services profiles are active by default, so to build an ear containing all services, just execute:
+All services profiles are active by default, so to build an ear containing all services, except for Direct, X12 just execute:
```
Comment (christophermay07) [inline README.md]: Suggestion, possibly change "an ear" to "a GlassFish ear" (since that's the only one built by default).

File: README.md [hunk 4]
```diff
@@ -97,57 +104,16 @@ You can also specify explicitly what services are included in the ear by passing
...
-These steps will install and configure a Glassfish Application Server to deploy and use CONNECT. Lets get started.
-
-Navigate to the <CONNECT_CLONE_DIR>/Product/Install directory
-
-        $ cd <CONNECT_CLONE_DIR>/Product/Install
-
-Next, copy install.properties to local.install.properties and update with your local information. Generally this is just specifying a password for mysql.
-
-        $ cp install.properties local.install.properties
+For some application server deployments the generated .ear needs different dependencies. Follow below wiki links for CONNECT supported application server build and deployment steps.
```
Comment (christophermay07) [inline README.md]: Consider changing "below wiki links" to "the wiki links below".

File: README.md [hunk 5]
```diff
@@ -97,57 +104,16 @@ You can also specify explicitly what services are included in the ear by passing
...
-
-        $ cp install.properties local.install.properties
-
-Lastly, we're going to install the Glassfish Application Server
-
-        $ ant install
+  * Setup GlassFish, MySQL & Deploy CONNECT:[Glassfish Install](https://connectopensource.atlassian.net/wiki/display/CONNECT4/Building+CONNECT+4.4+from+Source) 
+  * WebSphere, MySQL & Deploy CONNECT:[WebSphere Setup](https://connectopensource.atlassian.net/wiki/pages/viewpage.action?pageId=17203313)
+  * JBoss7, MySQL & Deploy CONNECT:[JBoss7 Setup](https://connectopensource.atlassian.net/wiki/pages/viewpage.action?pageId=26214665)
+  * WebLogic, MySQL & Deploy CONNECT:[WebLogic Setup](https://connectopensource.atlassian.net/wiki/pages/viewpage.action?pageId=17203304)
```
Comment (christophermay07) [inline README.md]: Suggestion: "MySQL" and "Deploy CONNECT" are implied by the sentence above this list, so you could just use:
- GlassFish: (link)
- WebSphere: (link)

etc.

(Also, add a space between the colon and the link, so it's not run together)

File: README.md [hunk 6]
```diff
@@ -206,7 +166,7 @@ Alternatively, any of these properties can be set in your maven settings.xml fil
 ###Run the Validation Suite via SoapUI
 The Validation Suite can be run with SoapUI. First, follow the instructions "Setting up SoapUI" below.
 
-Set the property "GatewayPropDir" in ConnectValidation-soapui-project.properties in the Validation Suite directory. This should be set to the gateway configuration directory. For GlassFish this is <GlassFish home>/domains/domain1/config/nhin; there is an equivalent in WebSphere
+Set the property "GatewayPropDir" in ConnectValidation-soapui-project.properties in the Validation Suite directory. This should be set to the gateway configuration directory. For GlassFish this is <GlassFish home>/domains/domain1/config/nhin; there is an equivalent in WebSphere,JBoss and WebLogic.
```
Comment (christophermay07) [inline README.md]: Add a space before "JBoss"

**Comment by christophermay07** [comment]
There's a "Usage" section, with "TK" as the only text.  This should be removed if we're not using it.

**Comment by christophermay07** [comment]
Not shown in GitHub diff: Typo "applicaiton" should be "application"

**Comment by christophermay07** [comment]
The raw README.md (NOT the parsed version) has a bunch of links at the bottom that appear to be unused; can they be removed?

**Comment by christophermay07** [comment]
The link for "Setting up CONNECT as a Direct HISP" should be relative (like the one used for HISTORY.md), not absolute.

**Comment by christophermay07** [comment]
"By advancing the adoption of interoperable health IT systems and health information exchanges, the country will better be able to achieve the goal of making sure **all citizens have electronic health records by 2014.**" (emphasis mine) - is that still relevant?

File: README.md [hunk 7]
```diff
@@ -39,11 +43,14 @@ Before you get started, you'll need the following installed and set up:
...
-* [Eclipse Juno](http://www.eclipse.org/downloads/)
-  * [egit plugin](http://www.eclipse.org/egit/)
-  * [m2eclipse plugin](http://www.eclipse.org/m2e/download/)
+* Application Server (Below are Supported Servers)
+  * [GlassFish v3.1.2.2]
+(https://glassfish.java.net/downloads/3.1.2.2-final.html)
+  * [WebSphere Enterprise 8.5.5.3](http://www.ibm.com/developerworks/downloads/ws/wasdevelopers/)
+  * [WebLogic 12c (12.1.1)](http://www.oracle.com/technetwork/middleware/ias/downloads/wls-main-097127.html)
+  * [JBoss 7.1.1 Final](http://jbossas.jboss.org/downloads/)
 * [Apache Ant v1.7.1](http://ant.apache.org/)
```
Comment (christophermay07) [inline README.md]: Is there a specific reason we're leaving v1.7.1 on this list, or should we move up to 1.9.4 (or remove the version altogether, or change to "(latest)")?

File: README.md [hunk 8]
```diff
@@ -161,29 +126,19 @@ The certificates that come in the CONNECT source do not include a Certificate Re
...
 Testing
 -------
 ###Run the Validation Suite as part of install
 At the end of the mvn install process, an embedded GlassFish instance will start and the Validation Suite will run against it. The maven scripts automatically stand up the embedded glassfish using trust chain certificates:
 
-        $ cd <CONNECT_CLONE_DIR>/Product/SoapUI_Test/ValidationSuite
+        $ cd <CONNECT_CLONE_DIR>/Product/SoapUI_Test/ValidationSuite at: 
         $ mvn clean install
 
 ###Run the Validation Suite via Maven sript
```
Comment (christophermay07) [inline README.md]: Typo: "sript" should be "script"

**Comment by christophermay07** [comment]
Under ValidationSuite test case parameters, we have the following: "Subscribe", "Notify", "Unsubscribe"

Are these still relevant?

File: README.md (hunk 7 above)
Comment (sailajaa) [inline README.md]: In CONNECT 4.4 installation instructions wiki, we are specifying v1.7.1, for consistency purpose leaving the version as v1.7.1

Comment (christophermay07) [inline README.md]: Then this brings us full-circle to the heart of FHAC-141: do we need this information here, if it's on the wiki?  What happens if we merge your PR, and I go in tomorrow and update the wiki to reflect that Ant 1.9.4 is the version most recently tested for 'ant install'? (which I guarantee it is, I used it this morning :smile: )

**Comment by jsmith2** [comment]
Couple of basic suggestions.
- Leave in basic maven build instructions.  Should be clear how to build the code from where you fork the code.
- For all other build/deploy instructions provide the top level link to our installations page. "For further build and deploy instructions go here: <link>".
- Remove the contribution piece.  We have a more thorough wiki on forking and doing pull requests in our wiki.  Replace with one sentence and a link.
- We should provide our open source license type in the read me (should be apparent to the user that forks/downloads the code what use they are allowed with it).

File: README.md [hunk 9]
```diff
@@ -230,20 +154,10 @@ Then open your browser and view [file:///tmp/fullsite/index.html]
...
-[ant 1.7.1]: http://archive.apache.org/dist/ant/binaries/apache-ant-1.7.1-bin.zip
-[egit plugin]: http://www.eclipse.org/egit/download/
-[m2eclipse plugin]: http://eclipse.org/m2e/
-[MySQL 5.1.x]: http://dev.mysql.com/downloads/mysql/5.1.html
 
+Please checkout [code contribution](https://connectopensource.atlassian.net/wiki/display/CONNECTWIKI/How+to+Contribute+Code) for guidelines about how to contribute.
+
+License
+-------
+
```
Comment (jsmith2) [inline README.md]: This license link is wrong, use: https://connectopensource.atlassian.net/wiki/display/CONNECTWIKI/CONNECT+License+Agreement

Also move the license up to below the CONNECT description.

File: README.md [hunk 10]
```diff
@@ -27,10 +27,14 @@ Three primary elements make up the CONNECT solution:
...
+* 4.3 released March 2014
+* 4.2 released August 2013
+* 4.1 released April 2013
 * 4.0 released February 2013
 * 3.3 released March 2012
 
-For more information, about CONNECT's history see [HISTORY.md](./CONNECT/HISTORY.md)
+For more information about CONNECT's history, see [HISTORY.md](./HISTORY.md)
 
 Getting Started
```
Comment (jsmith2) [inline README.md]: Everything in the getting started section can be removed, except for a quick summary of the build steps.  Everything else is covered by a link to the base installation instructions.

File: README.md [hunk 11]
```diff
@@ -230,20 +154,10 @@ Then open your browser and view [file:///tmp/fullsite/index.html]
...
-[Download Maven]: http://maventest.apache.org/download.html
-[Install Maven]: http://maventest.apache.org/download.html#Installation
-[Eclipse]: http://www.eclipse.org/downloads/
-[ant 1.7.1]: http://archive.apache.org/dist/ant/binaries/apache-ant-1.7.1-bin.zip
-[egit plugin]: http://www.eclipse.org/egit/download/
-[m2eclipse plugin]: http://eclipse.org/m2e/
-[MySQL 5.1.x]: http://dev.mysql.com/downloads/mysql/5.1.html
 
+Please checkout [code contribution](https://connectopensource.atlassian.net/wiki/display/CONNECTWIKI/How+to+Contribute+Code) for guidelines about how to contribute.
+
```
Comment (jsmith2) [inline README.md]: Please use the tiny links for all references, as those will remain consistent if the name of the page is changed.

File: README.md [hunk 12]
```diff
@@ -5,15 +5,20 @@ CONNECT is an open source software solution that supports health information exc
 
 This software solution was initially developed by federal agencies to support their health-related missions, but it is now available to all organizations and can be used to help set up health information exchanges and share data using nationally-recognized interoperability standards.
 
+License
+-------
+
+CONNECT is released under the: [License](https://connectopensource.atlassian.net/wiki/x/mQCD)
```
Comment (christophermay07) [inline README.md]: - Remove the colon after "the"
- Change "License" to "Revised BSD License"
- Add a period at the very end

File: README.md [hunk 13]
```diff
@@ -5,15 +5,20 @@ CONNECT is an open source software solution that supports health information exc
...
+
 Uses
 ----
 CONNECT can be used to:
 
 * Set up a health information exchange within an organization
 * Tie a health information exchange into a regional network of health information exchanges using Nationwide Health Information Network standards
-* Send and receive Direct messages, see: [Setting up CONNECT as a Direct HISP](https://github.com/CONNECT-Solution/CONNECT/blob/CONNECT_integration/Product/Production/Services/DirectCore/README.md)
+* Send and receive Direct messages, see: [Setting up CONNECT as a Direct HISP](/Product/Production/Services/DirectCore/README.md)
```
Comment (christophermay07) [inline README.md]: Change the comma to a semicolon here.

File: README.md [hunk 14]
```diff
@@ -206,10 +97,12 @@ Alternatively, any of these properties can be set in your maven settings.xml fil
 ###Run the Validation Suite via SoapUI
 The Validation Suite can be run with SoapUI. First, follow the instructions "Setting up SoapUI" below.
 
-Set the property "GatewayPropDir" in ConnectValidation-soapui-project.properties in the Validation Suite directory. This should be set to the gateway configuration directory. For GlassFish this is <GlassFish home>/domains/domain1/config/nhin; there is an equivalent in WebSphere
+Set the property "GatewayPropDir" in ConnectValidation-soapui-project.properties in the Validation Suite directory. This should be set to the gateway configuration directory. For GlassFish this is <GlassFish home>/domains/domain1/config/nhin; there is an equivalent in WebSphere, JBoss and WebLogic.
 
 Run the Validation Suite project file ConnectValidation-soapui-project.xml via SoapUI's command line runner testrunner.sh (or testrunner.bat in Windows).
 
+You can find more details at: [Connect Validation Suite](https://connectopensource.atlassian.net/wiki/x/I4Ch)
```
Comment (christophermay07) [inline README.md]: Change "Connect" to "CONNECT" on this line

**Comment by christophermay07** [comment]
I've reviewed this PR, endorsing for merge without further comment.

**Pull request merged**